macaroonbakery/httpbakery/agent.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2017 Canonical Ltd.
+# Licensed under the LGPLv3, see LICENCE file for details.
+
+import base64
+import json
+
+import nacl.public
+import nacl.encoding
+import requests.cookies
+import six
+from six.moves.urllib.parse import urlparse
+
+
+def load_agent_file(filename):
+    """ load_agent_file loads agent information from the specified file.
+
+        The agent information loaded is returned as a cookie jar and a key,
+        these can be passed directly as the cookies and key parameter to
+        BakeryAuth.
+    """
+
+    with open(filename) as f:
+        data = json.load(f)
+    key = nacl.public.PrivateKey(data['key']['private'],
+                                 nacl.encoding.Base64Encoder)
+    cookies = requests.cookies.RequestsCookieJar()
+    for agent in data['agents']:
+        u = urlparse(agent['url'])
+        value = {'username': agent['username'],
+                 'public_key': data['key']['public']}
```

Comment:
Changing the type of exception isn't going to result in a less incomprehensible stack trace.